ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandStore.java

```diff
@@ -186,6 +199,25 @@ public QueuedCommandStatus enqueueCommand(
     return restoreCommands;
   }
 
+  public CompletableFuture<Void> getConsumerPositionFuture(final long offset) {
+    final CompletableFuture<Void> future = new CompletableFuture<>();
+
+    final long consumerPosition = getConsumerPosition();
+    if (consumerPosition > offset) {
+      future.complete(null);
+    } else {
+      commandOffsetFutures.add(Pair.of(future, offset));
+    }
+    return future;
+  }
+
+  // returns the next offset to be consumed
```

Comment:
Avoid using commands like this if you can. Prefer encoding the command into the function name, e.g. `getNextConsumerOffset()`.  Or maybe just name it `getMaxOffsetConsumed()` and internally subtract 1 from the position.  This might back it more intuitive to thing about.

```
 // Does this read better?
if (getMaxOffsetConsumed() >= requestedOffset) {
   return CompletableFuture.completed(null);
}
```

don't mind really. just a thought.

Reply:
I think you mean "comments" here?

Reply:
I did indeed mean comment :D